File: app/src/main/java/com/parabola/newtone/mvp/view/FavouritesPlaylistView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface FavouritesPlaylistView extends MvpView {

    void setPlaylistChangerActivation(boolean activate);

    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);

    void setCurrentTrack(int trackId);

}

File: app/src/main/java/com/parabola/newtone/presentation/playlist/favourites/FavouritesPlaylistView.kt
package com.parabola.newtone.presentation.playlist.favourites

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface FavouritesPlaylistView : MvpView {
    fun setPlaylistChangerActivation(activate: Boolean)

    fun setCurrentTrack(trackId: Int)
    fun refreshTracks(tracks: List<Track>)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)
}
